src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -18,22 +37,27 @@ export const TodoFilter: React.FC = () => {
 
       <p className="control is-expanded has-icons-left has-icons-right">
         <input
+          value={filter.query}
           data-cy="searchInput"
           type="text"
           className="input"
           placeholder="Search..."
+          onChange={event => onChange(event.currentTarget.value)}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
         <span className="icon is-right" style={{ pointerEvents: 'all' }}>
           {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
+          {filter.query.length > 0 && (
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={() => dispatch(actions.setSearchQuery(''))}
```

Comment:
Consider creating handler function here

Reply:
it was here I substituted it with dispatch to reduce code